core/src/main/java/org/kohsuke/stapler/QueryParameter.java

```diff
@@ -79,5 +79,19 @@ public Object parse(StaplerRequest2 request, QueryParameter a, Class type, Strin
             }
             return convert(type, value);
         }
+
+        private String getQueryParameterFromQueryString(StaplerRequest2 request, String name) {
+            String queryString = request.getQueryString();
```

Comment:
I don't think this is right, `QueryParameter` is used to get body parameters as well.

Since we moved from `GET` to `POST` for validation and are moving autocompletion away as well this would break it

Reply:
I see that @jtnord said it was a violation of the contract but Jenkins depends on this behaviour so this change can't be made without a lot of work I would say

Reply:
>  so this change can't be made without a lot of work I would say

Should be limited to to a couple of javascript libraries that handle this?

if we are never going to change the behaviour the least we can do is to fix the declared contract to march reality and tell consumers to not use this when they also want to process the request body.

I have zero recollection of what was the trigger for me to find and file #416

Reply:
As of https://www.jenkins.io/changelog/2.285/ (2021)

Reply:
Plugins (generally) do not construct the validation, it's part of form binding taglibs from core?

Reply:
On the server side they ask for parameters in their methods

Reply:
Sure, my point was if the taglibs sent a HTTP POST with query parameters (application/x-www-form-urlencoded) (in addition or replacement of the multipart/form-data rather then this could be fixed.

Now, having thought about this some more, some of the query parameters could be new secrets which would then be exposed by HTTP access logs which would be bad, so following that suggestion would not be a "good thing".™️